release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,242 @@
+                releaseNote.releaseNoteText = matcher.group(1);
+            }
+        }
+
+        releaseNotesList.add(releaseNote);
+        repositoryMap.put(releaseNote.repository, releaseNotesList);
+        categoryMap.put(releaseNote.category, repositoryMap);
+        editionMap.put(releaseNote.edition, categoryMap);
+    }
+
+    void output() throws Exception {
+        System.out.printf("## Summary\n\n");
+        Arrays.stream(Edition.values())
+            .forEach(edition -> {
+                outputSections(edition);
+                System.out.println("");
+            });
+    }
+
+    void outputSections(Edition edition) {
+        Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap = editionMap.get(edition);
+        if (categoryMap == null || categoryMap.isEmpty()) return;
+
+        System.out.printf("## %s edition\n", edition.getEdition());
+        Arrays.stream(Category.values())
+            .forEach(category -> outputReleaseNotes(category, categoryMap.get(category)));
+    }
+
+    void outputReleaseNotes(Category category, Map<Repository, List<ReleaseNote>> repositoryMap) {
+        if (repositoryMap == null || repositoryMap.isEmpty()) return;
+
+        System.out.printf("### %s\n", category.getDisplayName());
+        Arrays.stream(Repository.values())
+            .forEach(repository -> {
+                List<ReleaseNote> releaseNotes = repositoryMap.get(repository);
+                if (releaseNotes != null && !releaseNotes.isEmpty()) {
+                    if (!repository.equals(Repository.DB))
+                        System.out.printf("#### %s\n", repository.getDisplayName());
+                    for (ReleaseNote rn : releaseNotes) {
+                        System.out.printf("- %s\n", rn.releaseNoteText);
+                    }
+                }
+            });
+    }
+
+    enum Edition {
+        COMMUNITY("Community"),
+        ENTERPRISE("Enterprise");
+
+        private final String edition;
+
+        Edition(String edition) {
+            this.edition = edition;
+        }
+
+        public String getEdition() {
+            return this.edition;
+        }
+    }
+    
+    enum Category {
+        ENHANCEMENT("Enhancements"),
+        IMPROVEMENT("Improvements"),
+        BUGFIX("Bug fixes"),
+        DOCUMENTATION("Documentation"),
+        MISCELLANEOUS("Miscellaneous");
+
+        private final String displayName;
+
+        Category(String displayName) {
+            this.displayName = displayName;
+        }
+
+        public String getDisplayName() {
+            return this.displayName;
+        }
+
+        public static Category getByName(String name) {
+            return Arrays.stream(Category.values())
+                    .filter(v -> v.name().toLowerCase().equals(name))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("Invalid name: " + name));
+        }
```

Comment:
It looks like this method is never used?

Reply:
Thank you for the point it out. I removed never used methods. Thank you.